share/functions/fish_git_prompt.fish

```diff
@@ -530,11 +534,16 @@ function __fish_git_prompt_informative_status
     set -l invalidstate (count (string match -r "U" -- $stagedFiles))
     set -l stagedstate (math $x - $invalidstate)
     set -l untrackedfiles (command git ls-files --others --exclude-standard | wc -l | string trim)
+    set -l stashstate 0
+    set -l stashfile "$argv[1]/logs/refs/stash"
+    if set -q __fish_git_prompt_showstashstate; and test -e "$stashfile"
+        set stashstate (wc -l < "$stashfile" | string trim)
```

Comment:
`wc -l $stashfile` is actually quite a bit faster (relatively) than the redirection. Of course that'll also print the filename, but that can be avoided by changing the `string`:

```fish
set stashstate (wc -l $stashfile | string match -r '\d+')
```

There are some `wc`s that, for some reason, indent the number, so you'd get output like

```
    32 .git/logs/refs/stash
```

but that `string` can deal with it. It'll match the first number on every line and print just that (no "-e").

This runs in 3ms vs 7ms for the `string trim` version.

Reply:
Awesome!